app/db/index.js

```diff
@@ -29,26 +27,30 @@ function getAllMarkets() {
  * This returns a Promise so that once the database values are secured, they can
  * be worked with in a different location/file.
  */
-function getSpecificMarket(market) {
+function getSpecificMarket(name) {
   return new Promise((resolve, reject) => {
-    db.child(market).once('value')
-      .then((result) => resolve(result.val()))
+    Market
+      .findOne({ _id: name })
+      .then((result) => resolve(result))
       .catch((err) => reject(err));
   });
 }
 
+function generateKey(name, address) {
+  // Make sure illegal characters removed from key.
+  return (`${name}, ${address}`).replace(/[^0-9a-zA-Z, ]/gi, '').trim();
+}
+
 /**
  * If the user selected "NEW MARKET" on the assessment page, this method is
  * called to upload the market to the database.
  * @param {*} info All question answers from the assessment page.
  */
 function addNewMarket(info) {
-  let marketName = `${info.marketInfo.marketName}, ${info.marketInfo.address}`;
-  // Make sure illegal characters removed from key.
-  marketName = marketName.replace(/[^0-9a-zA-Z, ]/gi, '').trim();
+  const marketName = generateKey(info.marketInfo.marketName, info.marketInfo.address);
 
-  // Add a new child to the markets reference in the database.
-  db.child(marketName).set({
+  const insert = {
```

Comment:
we don't do anything with this object besides calling it once as a parameter so let's just inline it
```javascript
Market.create({
  _id: marketName,
  personalInfo: { ... },
  ...
});
```